packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -103,39 +103,44 @@ export const TranslationSidebar = forwardRef<
         );
       }
     }, [word.id, notesQuery]);
+    const {
+      isLoading: isSavingTranslatorNote,
+      mutateAsync: mutateTranslatorNote,
+    } = useMutation({
+      mutationFn: (data: { wordId: string; language: string; note: string }) =>
+        apiClient.words.updateTranslatorNote(data),
+      onSuccess: () => notesQuery.refetch(),
+    });
 
     const saveTranslatorNote = useMemo(
       () =>
         throttle(
-          async (noteContent: string) => {
-            await apiClient.words.updateTranslatorNote({
-              wordId: word.id,
-              language,
-              note: noteContent,
-            });
-            notesQuery.refetch();
-          },
+          (note: string) =>
+            mutateTranslatorNote({ wordId: word.id, language, note }),
           15000,
           { leading: false, trailing: true }
         ),
-      [language, notesQuery, word.id]
+      [language, mutateTranslatorNote, word.id]
     );
 
+    const { isLoading: isSavingFootnote, mutateAsync: mutateFootnote } =
+      useMutation({
+        mutationFn: (data: {
+          wordId: string;
+          language: string;
+          note: string;
+        }) => apiClient.words.updateFootnote(data),
```

Comment:
@arrocke Would this be a good place to use `useCallback`? What about for `onSuccess`?

Reply:
I don't think it's necessary here